slam-libraries/viam-cartographer/src/slam_service/config_test.cc

```diff
@@ -334,6 +334,58 @@ BOOST_AUTO_TEST_CASE(ParseAndValidateConfigParams_valid_config_no_camera) {
     delete argv;
 }
 
+BOOST_AUTO_TEST_CASE(
+    ParseAndValidateConfigParams_valid_offline_config_no_delete_processed_data) {
+    ResetFlagsForTesting();
+    std::vector<std::string> args{
+        "carto_grpc_server",  "-config_param={mode=2d}",
+        "-data_dir=/path/to", "-port=localhost:0",
+        "-sensors=",          "-map_rate_sec=60",
+        "-data_rate_ms=200"};
+    int argc = args.size();
+    char** argv = toCharArrayArray(args);
+    SLAMServiceImpl slamService;
+    ParseAndValidateConfigParams(argc, argv, slamService);
+    BOOST_TEST(slamService.offline_flag == true);
+    BOOST_TEST(slamService.delete_processed_data == false);
+    delete argv;
+}
+
+BOOST_AUTO_TEST_CASE(
+    ParseAndValidateConfigParams_valid_online_config_no_delete_processed_data) {
+    ResetFlagsForTesting();
+    std::vector<std::string> args{
```

Comment:
My understanding is that after the flag is changed to a bool, it will be required, so I think the flag should be included here.